packages/vaex-arrow/vaex_arrow/dataset.py

```diff
@@ -0,0 +1,81 @@
+__author__ = 'maartenbreddels'
+import logging
+
+import numpy as np
+import pyarrow as pa
+
+import vaex.dataset
+import vaex.file.other
+logger = logging.getLogger("vaex_arrow")
+
+
+class DatasetArrow(vaex.dataset.DatasetLocal):
+    """Implements storage using arrow"""
+
+    def __init__(self, filename=None, table=None, write=False):
+        super(DatasetArrow, self).__init__(name=filename, path=filename, column_names=[])
+        self._write = write
+        if table is None:
+            self._load()
+        else:
+            self._load_table(table)
+
+    def _load(self):
+        source = pa.memory_map(self.path)
+        reader = pa.open_stream(source)
+        table = pa.Table.from_batches([b for b in reader])
+        self._load_table(table)
+    
+    def _load_table(self, table):
+        self._length_unfiltered =  self._length_original = table.num_rows
+        for col in table.columns:
+            name = col.name
+            # TODO: keep the arrow columns, and support and test chunks
+            arrow_array = col.data.chunks[0]
+            arrow_type = arrow_array.type
+            buffers = arrow_array.buffers()
+            assert len(buffers) == 2
+            mask = None
+            bitmap_buffer = buffers[0]
+            data_buffer = buffers[1]
+            if bitmap_buffer is not None:
+                # arrow uses a bitmap https://github.com/apache/arrow/blob/master/format/Layout.md
+                bitmap = np.frombuffer(bitmap_buffer, np.uint8, len(bitmap_buffer))
+                # we do have to change the ordering of the bits
+                mask = 1-np.unpackbits(bitmap).reshape((len(bitmap),8))[:,::-1].reshape(-1)[:len(arrow_array)]
```

Comment:
Is this just changing the ordering or also doing the conversion from bit to byte mask?

Reply:
Yeah, arrow seems to use this layout https://github.com/apache/arrow/blob/master/format/Layout.md#null-bitmaps which is different than numpy's (un)packbits. Would be nice if pyarrow provides a function for this (or numpy gets extended)